xarray/core/variable.py

```diff
@@ -216,7 +217,16 @@ def _possibly_convert_objects(values):
     * pd.Timedelta
     """
     as_series = pd.Series(values.ravel(), copy=False)
+    # For why we need this behavior: https://github.com/pandas-dev/pandas/issues/61938
     result = np.asarray(as_series).reshape(values.shape)
+    if (
```

Comment:
Can we move this after the `writable` bit below instead?

Reply:
great idea, would be great to avoid a double `copy`